packages/cyberstorm/src/components/Forms/CreateTeamForm/CreateTeamForm.tsx

```diff
@@ -0,0 +1,111 @@
+"use client";
+import { faArrowsRotate } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useController, useForm, Form } from "react-hook-form";
+import * as Button from "../../Button";
+import { TextInput } from "../../TextInput/TextInput";
+import { useToast } from "../../Toast/Provider";
+import styles from "./CreateTeamForm.module.css";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { isRecord } from "../../../utils/type_guards";
+
+interface errorResponse {
+  error: { message: string };
+}
+
+function isErrorResponse(response: unknown): response is errorResponse {
+  return (
+    isRecord(response) &&
+    isRecord(response.error) &&
+    typeof response.error.message === "string"
+  );
+}
+
+export function CreateTeamForm() {
+  const toast = useToast();
+
+  interface formFields {
+    teamName: string;
+  }
+
+  const schema = z.object({
+    teamName: z.string({ required_error: "Team name is required" }),
```

Comment:
[Zod's refine](https://zod.dev/?id=refine) and regexp could perhaps be used to validate the value matches the rules we show on the form? While not a blocker per se, it would be nice to have this kind of "essentials" done right from the get-go.

Reply:
Agreed with Mythic to not do extensive client-side validation right now.